tornado/util.py

```diff
@@ -193,7 +193,11 @@ def exec_in(code, glob, loc=None):
 if PY3:
     exec("""
 def raise_exc_info(exc_info):
-    raise exc_info[1].with_traceback(exc_info[2])
+    try:
```

Comment:
Why did this only change in the python 3 version? Doesn't the py2 version have the same circular reference?

Reply:
@bdarnell Any answer to your question? I'm getting memory leaks (python 2.7)

Reply:
I put my best guess in https://github.com/tornadoweb/tornado/pull/1936#issuecomment-289306175. Please do not post comments on long-closed issues. Normally I'd say you should open a new issue with details about your situation, but since tornado's support for python 2.7 ended in 2019, I'm afraid you're on your own now.

Reply:
I tried adding the finally statement - didn't work. I tried upgrading python 2.7.12->2.7.17 - didn't work. Are the memory leak issues solved in python3 for Tornado?